### PR TITLE
NMS-12692: Do not use hostname in keys

### PR DESCRIPTION
**Pull request by fooker**
The hostnames are transported as side-inputs instead.

JIRA: https://issues.opennms.org/browse/NMS-12692

### CODE REVIEW

**Comment by fooker** [comment]
This just uses the first available hostname if the change. (Whereas "first" is not temporal fist but the first one encountered while grouping)

File: main/src/main/java/org/opennms/nephron/Pipeline.java [hunk 1]
```diff
@@ -424,6 +449,17 @@ public void processElement(ProcessContext c) throws JsonProcessingException {
         });
     }
 
+    private static ParDo.SingleOutput<FlowDocument, KV<String, String>> extractHostname() {
+        return ParDo.of(new DoFn<FlowDocument, KV<String, String>>() {
+            @ProcessElement
+            public void processElement(ProcessContext c) {
+                final FlowDocument flowDocument = c.element();
+                Optional.ofNullable(flowDocument.getSrcHostname()).ifPresent(hostname -> c.output(KV.of(flowDocument.getSrcAddress(), hostname)));
```
Comment (j-white) [inline main/src/main/java/org/opennms/nephron/Pipeline.java:457]: May want to check for empty strings as well. I recall having to deal with these in the proto3 payloads.

File: main/src/test/java/org/opennms/nephron/FlowAnalyzerTest.java [hunk 2]
```diff
@@ -92,10 +100,13 @@ public void canComputeTotalBytesInWindow() {
         }
         TestStream<FlowDocument> flowStream = flowStreamBuilder.advanceWatermarkToInfinity();
 
+        final PCollectionView<Map<String, String>> hostnames = p.apply(Create.<KV<String, String>>of(KV.of("1.2.3.4", "example.com")))
```
Comment (j-white) [inline main/src/test/java/org/opennms/nephron/FlowAnalyzerTest.java:103]: Would be good to also add test coverage to verify that the hostnames end up in the `FlowSummary` objects as expected.

**Review by j-white — COMMENTED** [review]
👍 looks like this should work

**Review by christianpape — APPROVED** [review]
Nice!

**Review by j-white — APPROVED** [review]

**Pull request merged**